src/priceFeeder/substrate/feeder.ts

```diff
@@ -0,0 +1,66 @@
+import { ApiPromise } from '@polkadot/api';
+import { WsProvider, HttpProvider } from '@polkadot/rpc-provider';
+import Keyring from '@polkadot/keyring';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { stringToU8a } from '@polkadot/util';
+import BigNumber from 'bn.js';
+
+import logger from '../../logger';
+import { FeederKind, Listing } from '../types';
+
+const createProvider = (endpoint: string): WsProvider | HttpProvider => {
+  if (endpoint.startsWith('wss') || endpoint.startsWith('ws')) {
```

Comment:
`endpoint.startsWith('ws')` also covers `wss` case, same for http & https
if you want to be explicit, can just add a comment